test/mocks.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library analyzer_cli.test.mocks;
+
+import 'package:analyzer/analyzer.dart';
+import 'package:analyzer/src/generated/engine.dart';
+import 'package:analyzer/src/generated/source.dart';
+import 'package:analyzer_cli/src/options.dart';
+import 'package:mockito/mockito.dart';
+
+
+class MockAnalysisError extends Mock implements AnalysisError {
```

Comment:
I'm not sure how much sense it makes to create mocks for light-weight objects, like AnalysisError (and several others), but I don't have a strong objection to it either.